### PR TITLE
fix(destination): facebook pixel userData capturing fix for both identified and annonymous users

### PR DESCRIPTION
**Pull request by shrouti1507**
## PR Description

our device mode Facebook Pixel integration, did not send External ID for our Track events. But for our CAPI events does. Therefore, customers are getting pinged from Facebook with an alert saying that "100% of their CAPI events are coming in with external_id while 0% of their Pixel events are"
This is important for anonymous traffic and for event match quality for Facebook.

For sending user data the only way is to send it via init call. and it is strongly recommended to keep it inside the base code.

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/790)

### CODE REVIEW

File: src/integrations/FacebookPixel/constants.js [hunk 1]
```diff
@@ -69,4 +69,18 @@ const traitsMapper = [
   },
 ];
 
-export { traitsMapper, CNameMapping, NAME };
+const reserve = [
```
Comment (utsabc) [inline src/integrations/FacebookPixel/constants.js:72]:
```suggestion
const reserveTraits = [
```

Comment (shrouti1507) [reply src/integrations/FacebookPixel/constants.js]: addressed

**Review by utsabc — APPROVED** [review]

**Comment by utsabc** [comment]
Lets add unit test basic ones atleast

**Comment by bardisg** [comment]
get latest, have security issues fixed

**Review by saikumarrs — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by MoumitaM — APPROVED** [review]

**Pull request merged**